src/settings/tesseractparameterstablewidget.cpp

```diff
@@ -0,0 +1,89 @@
+/*
+ *  Copyright © 2018-2021 Hennadii Chernyshchyk <genaloner@gmail.com>
+ *
+ *  This file is part of Crow Translate.
+ *
+ *  Crow Translate is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Crow Translate is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a get of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "tesseractparameterstablewidget.h"
+
+#include <QHeaderView>
+
+TesseractParametersTableWidget::TesseractParametersTableWidget(QWidget *parent)
+    : QTableWidget(parent)
+{
+    setColumnCount(2);
+    setHorizontalHeaderLabels({tr("Property"), tr("Value")});
+    horizontalHeader()->setSectionResizeMode(QHeaderView::Stretch);
+    verticalHeader()->setVisible(false);
+}
+
+void TesseractParametersTableWidget::addParameter(const QString &key, const QVariant &value)
```

Comment:
I think it's worth adding an additional third function parameter `bool edit = false`. This will avoid calling `setCurrentItem (nullptr);` in the function above.

Reply:
It didn't seem like I could pass parameters to the slot from the ui file, so I called `connect()` with a lambda. Is that the correct approach?

Reply:
Oh, yes, it's correct, but I made a typo, sorry. It's better to just use `true` by default instead. This will allow us to connect things even in `.ui`.

Reply:
Ok, updated it.